packages/api/src/router/post.ts

```diff
@@ -6,4 +8,42 @@ export const postRouter = createTRPCRouter({
       orderBy: { createdAt: "desc" },
     });
   }),
+  createPost: protectedProcedure
+    .input(z.object({ title: z.string(), contents: z.string() }))
```

Comment:
I think that both the title and contents should be non-empty. Might also want to set a maximum length for the title

Reply:
I wanted to clarify, does that mean whitespaces should be invalid? ie if i just make the title a space it should not allow me to execute the endpoint right?

Reply:
If you do `z.string().trim().min.....` it will remove the whitespace from the string before checking the length

Reply:
I just updated the branch, lmk what u think

Reply:
yep looks good